src/index.js

```diff
@@ -0,0 +1,159 @@
+export default class toColor {
+  HUE_MAX = 360;
+
+  constructor(seed, options) {
+    this.options = options || {};
+    this.seed = typeof seed === 'string' ? this._stringToInteger(seed) : seed;
+    this.known = [];
+  }
+
+  getColor() {
+    const { distance, knownMax } = this.options;
+    const h = this._pickHue();
+    const s = this._pickSaturation(h);
+    const b = this._pickBrightness(h, s);
+    const hsl = this._HSVtoHSL(h, s, b);
+    const PASSABLE_DISTANCE = typeof distance !== 'undefined' ? distance : 37; 
+    const KNOWN_MAX = typeof knownMax !== 'undefined' ? knownMax : 20; 
+
+    // Detect color similarity. If values are too close to one another, call
+    // getColor until enough dissimilarity is acheived.
+    if (this.known.length &&
+      this.known.length <= KNOWN_MAX &&
+      this.known.some(v => this._euclideanDistance(v, hsl) < PASSABLE_DISTANCE)) {
+      return this.getColor();
+    } else {
+      this.known.push([...hsl]);
+      return this._colorWithModifiers(h, s, b) 
+    }
+  }
+
+  _colorWithModifiers = (h, s, b) => {
+    const clamp = (n, min, max) => n <= min ? min : n >= max ? max : n;
+    const { brightness, saturation } = this.options;
+
+    // Modify brightness/saturation if provided
+    s = saturation ? clamp(s + saturation, 0, 100) : s;
+    b = brightness ? clamp(b + brightness, 0, 100) : b;
+
+    // re-run conversion accounting for post modifications to s and b.
+    const hsl = this._HSVtoHSL(h, s, b);
+
+    return {
+      hsl: {
```

Comment:
Why not flatten this object and return raw_hsl and formatted_hsl? Seems like unecessary nesting.

Reply:
yeah I waffled on this but I wanted to future proof things in case we wanted to return other color spaces like lab. I guess instead of a `getColor()` method it could be something like `getHSL()` instead?